server/driver_tidb.go

```diff
@@ -391,11 +394,26 @@ func (trs *tidbResultSet) OnFetchReturned() {
 }
 
 func (trs *tidbResultSet) Columns() []*ColumnInfo {
+	if trs.columns != nil {
+		return trs.columns
+	}
+	// for prepare statement, try to get cached columnInfo array
+	if trs.preparedStmt != nil {
+		ps := trs.preparedStmt
+		if colInfos, ok := ps.ColumnInfos.([]*ColumnInfo); ok {
+			trs.columns = colInfos
+		}
+	}
 	if trs.columns == nil {
```

Comment:
we can remove this `if`~?

Reply:
No, when we cached info, but `Columns` for a statement is first called, `trs.columns` will not be nil at this line.